fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/FedoraResource.java

```diff
@@ -108,11 +109,23 @@
     Date getLastModifiedDate();
 
     /**
-     * Check if this object uses a given mixin
+     * Check if this object uses a given RDF type
+     *
+     * <p>Note: the type parameter should be in prefixed short form, so ldp:Container or ex:Image
+     * are both acceptable types. This method does not assume any jcr to fedora prefix mappings are
+     * managed by the implementation, so hasType("jcr:frozenNode") is a valid use of this method.</p>
+     *
      * @param type the given type
-     * @return a collection of mixin names
+     * @return whether the object has the given type
      */
     boolean hasType(final String type);
+
+    /**
+     * Get the RDF:type values for this resource
+     * @return a list of types for this resource
+     */
+    List<URI> getTypes();
```

Comment:
Sure, it's trivial. All I have to do is type some resource in the repo as an `rdfs:Class` and then use a relative URI. It turns out that it doesn't matter, though. At least since Java 7, `java.net.URI` can hold a relative URI. So no worries.